### DISCUSSION TITLE
Implement the density bar chart from stable mania editor back into lazer

### DISCUSSION THREAD

**Discussion by Hydria** [ppy/osu#21660] [Ideas]
Currently a lot of mappers in mania often use the density bar chart on the left hand of the editor as a way to judge the overall smoothness and consistency of a chart, and as a way to point out potential pain points or issues throughout a map whilst modding. It can also help new mappers by showing what is too much / too little relative to the song being mapped.

Having that missing currently means that people have to put in a lot more effort to find out the same information which can deter people from mapping in lazer. Plus, with the current horizontal note implementation at the top representing density it seems like so much more of a downgrade than before (aka pretty much unusable for mania mappers).

If possible, it would be nice to bring the bar chart back into lazer, maybe as a toggleable option for users (depending on client width), with the following additions:
- make the bar scale with SR instead of BPM to avoid some songs just becoming a pink wall due to low BPM
- add difficulty-related colours for certain bar lengths to represent how hard one of the sections actually is in comparison to another

**Comment by AncuL001** [comment]
could also be implemented to other modes using different layout, since the goal of density bar chart is also valid for them

**Comment by peppy** [comment]
We do plan on showing map density/difficulty-over-time in the editor, similar to during gameplay. It will use the same implementation.

**Comment by SQsuki** [comment]
您好，我是黄淑勤。您的邮件我已收到，感谢来信。
